frontend/src/utils/StockAPIClient.ts

```diff
@@ -0,0 +1,73 @@
+import {Stock, StockDetail} from "../models.ts";
+
+export const fetchStocksList = async (): Promise<Stock[]> => {
+    const response = await fetch('http://localhost:3000/api/v1/stocks', {
```

Comment:
issue(blocking): il faudrait que tu passes en fichier de configuration. Vite fonctionne de la façon suivante :

- Crée un fichier .env dans le même répertoire que le fichier de configuration vit.config.js
- Dedans crée une variable nommée  VITE_API_SERVER_URL
```Typescript
VITE_API_SERVER_URL=http://localhost:3000
```
- Exemple de code pour lire la valeur du fichier .env:
```Typescript
console.info('api server url', VITE_API_SERVER_URL);
```

Les variable dans le fichier .env doivent avoir leur nom préfixés par VITE_ si non, elles ne sont pas lues.

Voir la doc ici: https://vitejs.dev/guide/env-and-mode.html

Dans un second temps, tu peux faire une classe qui s'occupera de gérer la lecture depuis Vite. Mais déjà, si ça, ça fonctionne, ce sera beaucoup plus propre.

Reply:
done
A check voir si c'est correct